direct_bindings_api/ir.cpp

```diff
@@ -0,0 +1,406 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <direct_bindings.h>
+
+#include <fusion.h>
+#include <ir/base_nodes.h>
+
+namespace direct_bindings {
+
+// For all nodes, use multiple inheritance to disable destructor with
+// std::unique_ptr<Statement, py::nodelete>. This class will
+// disable memory management because it is handled automatically by IrContainer.
+
+namespace {
+
+using namespace nvfuser;
+
+void bindBaseNodes(py::module& direct_bindings) {
+  // Statement
+  py::class_<Statement, std::unique_ptr<Statement, py::nodelete>>(
+      direct_bindings, "Statement")
+      .def(
+          "name",
+          &Statement::name,
+          R"(
+Get the unique identifier of this statement.
+
+Returns
+-------
+int
+    The integer that represents this statement's unique identifier.
+)")
```

Comment:
as much as I appreciate the comment, the formatting make this file very hard to read.

Reply:
What do you propose then?

1. No docstrings
2. Avoid numpy-style docstrings for short summary only.
3. Move docstrings to python side.

Reply:
The docstrings are verbose in pytorch.
https://github.com/pytorch/pytorch/blob/main/torch/nn/modules/activation.py#L698-L724

Reply:
I think having doc string on python side might be cleaner to read, simply because how they are formatted in python vs as strings in c++.
However, if we can generate these bindings automatically, then I wouldn't mind having awkward comments/docstring all over the place. :stuck_out_tongue_winking_eye:  

Thanks for bearing with my nitpicking.